### ISSUE TITLE
Possible bug in operator() when using array as index 

### ISSUE THREAD

**Issue by awni** [arrayfire/arrayfire#2250]
This code throws an ArrayFire exception:

```
int main() {

  float a[] = {9, 8, 7, 6, 5, 4, 3, 2, 1};
  af::array A(9, a);

  float idx[] = {1, 3, 5};
  af::array IDX(3, idx);

  auto C  = A(IDX);
  af::print("", C);

  return 0;
}
```

Ouptput:
```
terminate called after throwing an instance of 'af::exception'
  what():  ArrayFire Exception (Device out of memory:101):
In function void* cuda::MemoryManager::nativeAlloc(size_t)
In file src/backend/cuda/memory.cpp:184
CUDA Error (2): out of memory


In function af::array::array_proxy::operator af::array()
In file src/api/cpp/array.cpp:719
```

Oddly if I change the line `auto C ...` to `af::array C ...` it seems to work fine. 

Any ideas?

**Comment by umar456** [comment]
This is happening because we are returning the array_proxy class to handle indexing operations. It implicitly converts to an array object but because you are using auto, C becomes a array_proxy object which should not happen because this is an implementation detail. I think print should be able to handle this situation. As a workaround you should avoid using auto when performing an indexing operation. Thanks for reporting this.

**Comment by awni** [comment]
This seems to be fixed as of v3.6.1!
